stumpy/cache.py

```diff
@@ -117,3 +145,69 @@ def _get_cache():
     site_pkg_dir = site.getsitepackages()[0]
     numba_cache_dir = site_pkg_dir + "/stumpy/__pycache__"
     return [f.name for f in pathlib.Path(numba_cache_dir).glob("*nb*") if f.is_file()]
+
+
+def _recompile():
+    """
+    Recompile all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    If the `numba` cache is enabled, this results in saving (and/or overwriting)
+    the cached numba functions to disk.
+    """
+    for module_name, func_name in get_njit_funcs():
+        module = importlib.import_module(f".{module_name}", package="stumpy")
+        func = getattr(module, func_name)
+        func.recompile()
+
+    return
+
+
+def _save():
+    """
+    Save all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
+    _enable()
+    _recompile()
+
+    return
+
+
+def save():
+    """
+    Save/overwrite all the cache data files of
+    all-so-far compiled njit functions.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
+    if numba.config.DISABLE_JIT:
+        msg = "Cannot save cache because NUMBA JIT is disabled"
+        raise OSError(msg)
```

Comment:
Is it possible to avoid raising an error here and, instead:

```
if numba.config.DISABLE_JIT:
    msg = "Could not save/cache function because 'NUMBA_DISABLE_JIT=True'"
    warnings.warn(msg)
else:
    warnings.warn(CACHE_WARNING)
    _save()
```

Whenever possible, we should avoid raising an error unless it is impossible to proceed.

Reply:
> avoid raising an error unless it is impossible to proceed.

Noted. 

Should we check if the env variable `NUMBA_DISABLE_JIT` exist before saying `because 'NUMBA_DISABLE_JIT=True'`? Because one can just change the value of `numba.config.DISABLE_JIT` without creating the env var `NUMBA_DISABLE_JIT`.

Reply:
> `NUMBA_DISABLE_JIT` can affect `numba.config.DISABLE_JIT` but never the other way around.

Correct. I assume we are going to ignore the following possibility:

>  if a user explicitly sets numba.config.DISABLE_JIT = True

Because otherwise, the warning message might be confusing. Right? I mean the if condition can be satisfied without having the env var `NUMBA_DISABLE_JIT`. That is why I mentioned if we should check the env var as well here (or, revise the warning message (?))

Reply:
Gotcha! Yes, we should use your original message instead of referring to `NUMBA_DISABLE_JIT` then (but don't raise an error).